js/main.js

```diff
@@ -452,6 +525,219 @@ function handleImageClick(event) {
+        if (paramVal === 'null') {
+          continue;
+        }
+        for (j = 0; j < metadata.likedMetadata.geoLocation[currentSearchParam][paramVal]; j++) {
+          fullValues.push(paramVal);
+        }
+      }
+    } else if (Object.keys(metadata.likedMetadata.date).includes(currentSearchParam)) {
+      for (paramVal in metadata.likedMetadata.date[currentSearchParam]) { // e.g. paramVal = '1842'
+        if (paramVal === 'null') {
+          continue;
+        }
+        for (j = 0; j < metadata.likedMetadata.date[currentSearchParam][paramVal]; j++) {
+          fullValues.push(paramVal);
+        }
+      }
+    } else {
+      for (paramVal in metadata.likedMetadata[currentSearchParam]) { // e.g. paramVal = 'Vincent van Gogh'
+        if (paramVal === 'null') {
+          continue;
+        }
+        for (j = 0; j < metadata.likedMetadata[currentSearchParam][paramVal]; j++) { // e.g. ...[paramVal] = 23
+          fullValues.push(paramVal);
+        }
+      }
+    }
+    // Create an object with keys of the metadata type and values of arrays of the unique metadata values to search
+    // e.g. {artistDisplayName = ['van Gogh', 'Monet'], medium = ['oil', 'watercolor'] }
+    searchVals[currentSearchParam] = [...new Set(shuffleArray(fullValues).slice(0, numVals))];
+  }
+  return searchVals;
+}
+
+function generateSearchURL(numProps, numVals) {
+  var valuesToSearch = getSearchValues(getSearchTerms(numProps), numVals);
+  var possibleGeoLocationProps = Object.keys(metadata.likedMetadata.geoLocation);
+  // number of years to look back if year is before...
+  var dateSearchRanges = {
+    0: 500,
+    1300: 200,
+    1800: 100,
+    1900: 50,
+    3030: 20
+  };
+
+  var searchURL = metEndpoint + 'search?hasImages=true';
+  var qQuery = '&q=*';
+
+  // currently using simplified solution with just 1 property, 1 value
+  for (var key in valuesToSearch) {
+    if (possibleGeoLocationProps.includes(key)) {
+      searchURL += '&geoLocation=' + valuesToSearch[key][0];
+    } else if (['artistDisplayName', 'culture'].includes(key)) {
+      searchURL += '&artistOrCulture=true';
+      qQuery = '&q=' + valuesToSearch[key][0];
+    } else if (key === 'medium') {
+      // handle complex medium values (can have multiple separated by commas)
+      var outputMediumStr = '';
+      var splitMediums = valuesToSearch[key][0].split(',');
+      // Stop at first three mediums, some works can have many more but they are too specific to search on
+      // API is expecting '...&medium=Silk|Metal|Pine&q=...
+      for (var i = 0; i < splitMediums.length && i < 3; i++) {
+        splitMediums[i] = splitMediums[i].replace(' ', '');
+        outputMediumStr += splitMediums[i][0].toUpperCase() + splitMediums[i].slice(1).toLowerCase();
+        outputMediumStr += '|';
+      }
+      searchURL += '&medium=' + outputMediumStr.slice(0, -1);
+    } else if (Object.keys(metadata.likedMetadata.date).includes(key)) {
```

Comment:
You can do `if(metadata.likedMetadata.date[key] !== undefined)` to save you doing a conversion to an array